app/user/api.py

```diff
@@ -1,39 +1,40 @@
-from . import service
-from .errors import UserNotFoundError, UserAuthenticationError
+from .service import create_user, delete_user
+from app.events.service import (
+    generate_account_signup_event,
+    generate_account_delete_event,
+)
+from .errors import UserNotFoundError, UserAlreadyExistsError
 from .user import User
 from flask import jsonify
 from firebase_admin import auth
-import jwt
+from firebase_admin.auth import UserRecord
+from typing import Dict, Any
+from app.auth import with_auth_user
 
-
-def get_user_profile(email):
+# Note: this endpoint will NOT use our middleware wrapper, since at this point we have no User record yet. We will default to the parameters connexion gives us
+def new_user_signup(token_info: Dict[str, Any]):
     try:
-        user = service.find_user(email)
-    except UserNotFoundError:
-        return {"error": "User not found"}, 404
+        firebase_user_record: auth.UserRecord = auth.get_user(token_info["uid"])
+        new_user = create_user(firebase_user_record.email)
+        generate_account_signup_event(new_user)
```

Comment:
The events creation should live in the service layer, not the API layer in my opinion. Saves the developer from worries about events if I'm calling user services.